src/syntax/Parser.js

```diff
@@ -2247,7 +2247,8 @@ export class Parser {
     let name = this.parsePropertyName_();
 
     if (this.options_.propertyMethods && this.peek_(OPEN_PAREN)) {
-      let fs = this.pushFunctionState_(FUNCTION_STATE_METHOD);
+      let fs = this.pushFunctionState_(FUNCTION_STATE_METHOD |
+          FUNCTION_STATE_DERIVED_CONSTRUCTOR);
```

Comment:
This change would break the following... I guess I missed a test for it?

```js
class C {
  m() {
    super();  // Should be SyntaxError
  }
}
```

I guess that means that we cannot use parsePropertyDefinition for this :(